File: go/internal/statex/state_commit_test.go
package statex

import (
	"strings"
	"testing"
)

func TestBuildCommitMessage(t *testing.T) {
	message := BuildCommitMessage("chore: working", "keep lease alive", []Trailer{
		{Key: "dwp-state", Value: "working"},
		{Key: "dwp-run-id", Value: "abc"},
	})

	expected := []string{
		"chore: working",
		"",
		"keep lease alive",
		"",
		"dwp-state: working",
		"dwp-run-id: abc",
		"",
	}
	if message != strings.Join(expected, "\n") {
		t.Fatalf("unexpected message:\n%s", message)
	}
}

File: js/test/stateCommit.test.js
import { test, expect } from 'vitest';
import { buildStateCommitMessage } from '../AgentsOrchestrator/stateCommit.js';

test('buildStateCommitMessage renders prompt and contiguous trailer block', () => {
  const message = buildStateCommitMessage('chore: working', 'keep lease alive', [
    { key: 'dwp-state', value: 'working' },
    { key: 'dwp-run-id', value: 'abc' }
  ]);

  expect(message).toBe([
    'chore: working',
    '',
    'keep lease alive',
    '',
    'dwp-state: working',
    'dwp-run-id: abc',
    ''
  ].join('\n'));
});
